internal/zinc-compile-core/src/main/scala/sbt/internal/inc/javac/DiagnosticsReporter.scala

```diff
@@ -64,85 +66,117 @@ final class DiagnosticsReporter(reporter: Reporter) extends DiagnosticListener[J
 
     import sbt.util.InterfaceUtil.problem
     val msg = fixedDiagnosticMessage(d)
-    val pos: xsbti.Position = new PositionImpl(d)
+    val pos: xsbti.Position = PositionImpl(d)
     if (severity == Severity.Error) errorEncountered = true
     reporter.log(problem("", pos, msg, severity))
   }
+}
 
-  private class PositionImpl(d: Diagnostic[_ <: JavaFileObject]) extends xsbti.Position {
-    // https://docs.oracle.com/javase/7/docs/api/javax/tools/Diagnostic.html
-    // Negative values (except NOPOS) and 0 are not valid line or column numbers,
-    // except that you can cause this number to occur by putting "abc {}" in A.java.
-    // This will cause Invalid position: 0 masking the actual error message
-    //     a/A.java:1: class, interface, or enum expected
-    private[this] def checkNoPos(n: Long): Option[Long] =
-      n match {
-        case NOPOS       => None
-        case x if x <= 0 => None
-        case x           => Option(x)
-      }
-
-    override val line: Optional[Integer] = o2jo(checkNoPos(d.getLineNumber) map { x =>
-      new Integer(x.toInt)
-    })
-    def startPosition: Option[Long] = checkNoPos(d.getStartPosition)
-    def endPosition: Option[Long] = checkNoPos(d.getEndPosition)
-    override val offset: Optional[Integer] = o2jo(checkNoPos(d.getPosition) map { x =>
-      new Integer(x.toInt)
-    })
-    override def lineContent: String = {
-      def getDiagnosticLine: Option[String] =
-        try {
-          // See com.sun.tools.javac.api.ClientCodeWrapper.DiagnosticSourceUnwrapper
-          val diagnostic = d.getClass.getField("d").get(d)
-          // See com.sun.tools.javac.util.JCDiagnostic#getDiagnosticSource
-          val getDiagnosticSourceMethod =
-            diagnostic.getClass.getDeclaredMethod("getDiagnosticSource")
-          val getPositionMethod = diagnostic.getClass.getDeclaredMethod("getPosition")
-          (Option(getDiagnosticSourceMethod.invoke(diagnostic)),
-           Option(getPositionMethod.invoke(diagnostic))) match {
-            case (Some(diagnosticSource), Some(position: java.lang.Long)) =>
-              // See com.sun.tools.javac.util.DiagnosticSource
-              val getLineMethod = diagnosticSource.getClass.getMethod("getLine", Integer.TYPE)
-              Option(getLineMethod.invoke(diagnosticSource, new Integer(position.intValue())))
-                .map(_.toString)
-            case _ => None
-          }
-        } catch {
-          // TODO - catch ReflectiveOperationException once sbt is migrated to JDK7
-          case ignored: Throwable => None
```

Comment:
Should we postpone this change for an RC cycle?

Reply:
You have a hawk eye. Sure, I'll be fine with moving that change to 1.1.x.

Reply:
It's clearer when ignoring whitespace changes: https://github.com/sbt/zinc/pull/465/files?w=1.